run_tests.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import os
+import signal
+from subprocess import check_call, Popen, check_output
+import sys
+import time
+import pip
+import urllib2
+import tarfile
+import shutil
+
+
+DIR_TEST_ENV = "test/env"
+DIR_JENKINS_ENV = 'jenkins-env'
+VERSION_VIRTUALENV = '1.9.1'
+
+os.environ["JENKINS_HOME"] = "jenkins-master"
+
+
+def python(*args):
+    pypath = [os.path.join(DIR_TEST_ENV, 'bin', 'python')]
+    check_call(pypath + list(args))
+
+
+def pip(*args):
+    pippath = [os.path.join(DIR_TEST_ENV, 'bin', 'pip')]
+    check_call(pippath + list(args))
+
+
+def download(url, destination):
+    response = urllib2.urlopen(url)
+    with open(destination, "wb") as code:
+        code.write(response.read())
+
+
+def kill_jenkins():
+    """Kill all Jenkins processes"""
+    print "Killing Jenkins"
+    pids = check_output(['lsof', '-i:8080', '-t'])
+    pids = pids.strip().split('\n')
+
+    for pid in pids:
+        os.kill(int(pid), signal.SIGKILL)
+
+
+def run_tests():
+    """Main Function that handles all the tests."""
+    try:
+        print "Checking Patches"
+        check_call('./test/check_patches.sh')
+
+        if os.path.exists("./" + DIR_JENKINS_ENV) and not os.environ.get('CI'):
+            print "Jenkins environment already exists!"
+            while True:
+                user_input = raw_input("Would you like to recreate it? ")
+                if user_input[0].lower() == "y":
+                    print "Running setup:"
+                    shutil.rmtree(DIR_JENKINS_ENV, True)
+                    check_call(["./setup.sh", DIR_JENKINS_ENV])
+                    break
+                elif user_input[0].lower() == "n":
+                    break
+                else:
+                    print "Please answer yes or no."
+        else:
+            print "Running setup"
+            check_call(["./setup.sh", DIR_JENKINS_ENV])
+
+        print "Starting Jenkins"
+        Popen(['./start.py', '>', 'jenkins.out'])
```

Comment:
So this is routing the stdout to the log file. It would be better to specify that as a separate kwarg in the Popen() call. See here for an example: http://stackoverflow.com/a/3190840